File: Lua/d1.lua
s = ... -- input
-- Part 1
print(({s:gsub('(','')})[2]-({s:gsub(')','')})[2])
-- Part 2
x=0
i=1
repeat
    x=x+(-1)^(s:sub(i,i)==')'and 1 or 0)
    i=i+1
until x<0
print(i)
-- Both parts
x=1
i=1
f=s:gmatch"."
for c in f do
    x=x+(1|-(c==')'and 1 or 0))
    if i then
        if x == 0 then
            print(i)
            i=false
        else i=i+1 end
    end
end
print(x-1)

File: C/d1.c
#include <stdio.h>
int main(){
    const char*s=""; /* input */
    /* Part 1 */ {
    const char*y=s;
    int x=0;
    for (char c=*y;c;c=*++y)x+=1|-(c-'(');
    printf("%d\n",x);
    } /* Part 2 */ {
    int x=1,i=0;
    for (;s[i];++i) {
        x+=1|-(s[i]-'(');
        if(!x){
            printf("%d\n",i);
            break;
        }
    }
    } /* Both parts */ {
    int y=1,x=1,i=0;
    for (;s[i];++i) {
        x+=1|-(s[i]-'(');
        if(y&!x){
            y=0;
            printf("%d\n",i);
        };
    }
    printf("%d\n",x-1);
}}
